src/main.ts

```diff
@@ -118,3 +114,17 @@ async function comment(body: string): Promise<void> {
     core.info(`Not a pull request, skipping comment`)
   }
 }
+
+async function generateBadgesAndCommit(
+  coveredProjects: CoveredProject[]
+): Promise<void> {
+  try {
+    core.info(`Configuring git...`)
+    await configureGit()
+    core.info('Updating and pushing coverage badge...')
+    await generateBadges(coveredProjects)
+    await commitAndPushChanges('chore: coverage badges [skip ci]')
+  } catch (error) {
+    core.warning(`Failed to commit and push coverage badge due to ${error}.`)
+  }
+}
```

Comment:
Does this actually commit on the PR ref now? It seems to want to commit to main for me, which fails since the main branch is protected

Reply:
We're currently using the code from [this fork](https://github.com/NEON-Software-Solutions/dart-coverage-assistant) and it works just fine. BUT we encountered another issue: the coverage badge file on every branch leads to a lot of tiring merge conflicts... Currently exploring better ways to solve this.